src/test/java/liuni/LiUniResourceTests.java

```diff
@@ -0,0 +1,344 @@
+
+            Response resp = resource.fetchTimeline();
+            testResult = resp.getStatus() == Response.Status.INTERNAL_SERVER_ERROR.getStatusCode();
+        }
+        catch (Exception e) {
+            testResult = false;
+        }
+
+        assertTrue(testResult);
+        reset(twitter);
+        reset(status);
+    }
+
+    @Test
+    public void testREST_fetchTimeline_fetchTimelineError_Entity() {
+        resource = new LiUniResource();
+        twitterTimeline = new TwitterTimeline();
+        twitterTimeline.setTwitter(twitter);
+        resource.setTwitterTimeline(twitterTimeline);
+
+        ResponseList<Status> respList = new ResponseListImpl<Status>();
+        respList.add(status);
+        boolean testResult;
+        String expectedError = (new ErrorModel()).getGeneralError();
+        try {
+            when(twitter.getHomeTimeline()).thenThrow(new TwitterException(""));
+
+            Response resp = resource.fetchTimeline();
+            testResult = resp.getEntity().toString().equals(expectedError);
+        }
+        catch (Exception e) {
+            testResult = false;
+        }
+
+        assertTrue(testResult);
+        reset(twitter);
+        reset(status);
+    }
+
+    /* Test LiUniResource .postTweet() REST method */
+    @Test
+    public void testREST_postTweet_successfulPost_Status() {
+        resource = new LiUniResource();
+        twitterStatus = new TwitterStatus();
+        twitterStatus.setTwitter(twitter);
+        resource.setTwitterStatus(twitterStatus);
+
+        boolean testResult;
+        String testString = "message";
+        try {
+            when(twitter.updateStatus(testString)).thenReturn(status);
+            when(status.getText()).thenReturn(testString);
+
+            Response resp = resource.postTweet(testString);
+            testResult = resp.getStatus() == Response.Status.CREATED.getStatusCode();
+        }
+        catch (Exception e) {
+            testResult = false;
+        }
+
+        assertTrue(testResult);
+        reset(twitter);
+        reset(status);
+    }
+
+    @Test
+    public void testREST_postTweet_successfulPost_Entity() {
+        resource = new LiUniResource();
+        twitterStatus = new TwitterStatus();
+        twitterStatus.setTwitter(twitter);
+        resource.setTwitterStatus(twitterStatus);
+
+        boolean testResult;
+        String testString = "message";
+        try {
+            when(twitter.updateStatus(testString)).thenReturn(status);
+            when(status.getText()).thenReturn(testString);
+
+            Response resp = resource.postTweet(testString);
+            testResult = resp.getEntity().toString().equals(testString);
+        }
+        catch (Exception e) {
+            testResult = false;
+        }
+
+        assertTrue(testResult);
+        reset(twitter);
+        reset(status);
+    }
+
+    @Test
+    public void testREST_postTweet_badPost_badTweetStatus() {
+        resource = new LiUniResource();
```

Comment:
since you keep using the same new LiUniResource and others on the tests, see about finding a more generic way of initializing them for all tests